hazelcast/src/main/java/com/hazelcast/client/impl/ClientEngineImpl.java

```diff
@@ -163,6 +167,22 @@ private Executor newExecutor() {
                 ExecutorType.CONCRETE);
     }
 
+    private Executor newClientQueryExecutor() {
```

Comment:
We figured out with @jerrinot that it may lead to deadlocks since the node may execute operations on other nodes.